public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,740 @@
+  const [enabled, setEnabled] = useState<boolean>(false);
+
+  const title = embeddable.getTitle();
+  const onAccordionToggle = (key) => {
+    const newAccordionsOpen = { ...accordionsOpen };
+    newAccordionsOpen[key] = !accordionsOpen[key];
+    setAccordionsOpen(newAccordionsOpen);
+  };
+  const onDetectorNameChange = (e, field) => {
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) =>
+      index <
+      (aggList.length < MAX_FEATURE_NUM ? aggList.length : MAX_FEATURE_NUM)
+  );
+
+  const getFeatureNameFromVisParams = (id) => {
+    let name = find(embeddable.vis.params.seriesParams, function (param) {
+      if (param.data.id === id) {
+        return true;
+      }
+    });
+
+    return name.data.label.replace(/[^a-zA-Z0-9-_]/g, '_');
+  };
+
+  const notifications = getNotifications();
+
+  const handleValidationAndSubmit = (formikProps) => {
+    if (!isEmpty(formikProps.errors)) {
+      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
+      notifications.toasts.addDanger('One or more input fields is invalid');
+    } else {
+      handleSubmit(formikProps);
+    }
+  };
+
+  const handleSubmit = (formikProps) => {
+    formikProps.setSubmitting(true);
+    try {
+      const detectorToCreate = formikToDetector(formikProps.values);
+      dispatch(createDetector(detectorToCreate))
+        .then(async (response) => {
+          notifications.toasts.addSuccess(
+            `Detector created: ${formikProps.values.name}`
+          );
+          dispatch(startDetector(response.response.id))
+            .then((startDetectorResponse) => {
+              notifications.toasts.addSuccess(
+                `Successfully started the real-time detector`
+              );
+            })
+            .catch((err: any) => {
+              notifications.toasts.addDanger(
+                prettifyErrorMessage(
+                  getErrorMessage(
+                    err,
+                    'There was a problem starting the real-time detector'
+                  )
+                )
+              );
+            });
+          enum VisLayerTypes {
+            PointInTimeEvents = 'PointInTimeEvents',
+          }
+          const fn = {
+            type: VisLayerTypes.PointInTimeEvents,
+            name: 'overlay_anomalies',
+            args: {
+              detectorId: response.response.id,
+            },
+          } as VisLayerExpressionFn;
+
+          const savedObjectToCreate = {
+            title: embeddable.vis.title,
+            pluginResourceId: response.response.id,
```

Comment:
Make sure to update the fields here like [this](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/487#discussion_r1202853023).